src/ucs/datastruct/ptr_array.h

```diff
@@ -120,13 +150,229 @@ void *ucs_ptr_array_replace(ucs_ptr_array_t *ptr_array, unsigned index, void *ne
                     (UCS_V_INITIALIZED(_var), 0) : \
                     !__ucs_ptr_array_is_free(_var = (void*)((_ptr_array)->start[_index])))
 
+/**
+ * Check if element is free, and calculate how many free elements are ahead.
+ *
+ * @param [in]     _elem  An element in the ptr array.
+ * @param [in/out] iter   Pointer to the foreach iterator, updated to next_idx.
```

Comment:
Pls, fix the documentation

Reply:
Fixed:
                  !__ucs_ptr_array_is_free(_var = (void*)((_ptr_array)->start[_index])))

/**
 * Check if element is free, and calculate how many free elements are ahead.
 *
 * @param [in]  _elem        An element in the ptr array.
 * @param [out] _size_elem   Number of free elements ahead if free, if not 1.
 *
 * @return 1 if the element is free and 0 if it's occupied.
 */